index_api-v2.md

```diff
@@ -0,0 +1,191 @@
+
+#### Output
+
+- `node_id` - hash of the `profile_url`
+
+> :construction: HASH ALGORITHM
+>
+> The current implementation of the index uses the SHA256 hashing algorithm which produces a 64-character output. The purpose of hashing the `profile_url` is to make it easy to reference as a path parameter when requesting information about the node from the index (e.g., `GET /nodes/{node_id}` as described below).
+
+#### Sequence Diagram - Add Node Profile to Index
+
+![Add Node Profile to Index](index_api-v1-diagram_1.png)
+
+### [`POST /nodes-sync`](https://app.swaggerhub.com/apis-docs/MurmurationsNetwork/IndexAPI/2.0.0#/Node%20Endpoints/post_nodes_sync)
+
+This endpoint is identical to the `POST /nodes-sync` endpoint above but it responds with an immediate success or failure status, so it is useful for posting nodes in an asynchronous call to the index.
+
+#### Input
+
+- Same as for the `POST /nodes` endpoint above
+
+#### Output
+
+- Same as for the `GET /nodes/{node_id}` endpoint below
+
+### [`GET /nodes/{node_id}`](https://app.swaggerhub.com/apis-docs/MurmurationsNetwork/IndexAPI/2.0.0#/Node%20Endpoints/get_nodes__node_id_)
+
+The record of a node in the index's database can be in one of five possible states: `received`, `validated`, `validation_failed`, `posted`, `post_failed` or `deleted`. The node will only be discoverable in the index when it has the status of `posted` or `deleted`.
+
+This endpoint enables a Murmurations profile generator to get and present an update to the node operator as to the status of the node profile after it has been submitted to the index (i.e., when using `POST /nodes`).
+
+#### Input
+
+- the `node_id` of the node's profile that is currently in the index
+
+#### Output
+
+Detailed response examples can be seen in the [Open API specification](https://app.swaggerhub.com/apis-docs/MurmurationsNetwork/IndexAPI/2.0.0#/Node%20Endpoints/get_nodes__node_id_) under the _Responses_ section for `200 OK` response code.
+
+##### Success
+
+- Confirmation of status (e.g., `received`, `validated`, `posted` or `deleted`)
+
+##### Error
+
+###### Validation Failed
+
+- Could not download profile from `profile_url`
+- Could not validate profile against one or more schemas
+- Could not find one or more schemas in library for validation
+
+###### Post Failed
+
+- Could not post validated profile to the index (internal service issue)
+
+### [`DELETE /nodes/{node_id}`](https://app.swaggerhub.com/apis-docs/MurmurationsNetwork/IndexAPI/2.0.0#/Node%20Endpoints/delete_nodes__node_id_)
+
+Node operators will use the `DELETE /nodes/{node_id}` endpoint to remove their profile from the index when they no longer want it listed.
+
+#### Input
+
+- the `node_id` of the profile that is currently in the index
+  - The profile must no longer be available at the `profile_url` (i.e., should return a `404 Not Found` error when accessing the URL) or it will not be removed from the index
```

Comment:
Because systems don't necessarily do a good job of returning 404s for missing resources (ie, when they're using URL rewrites to show a custom 404 page, or other redirection), I think `delete` requests should also take effect when the profile URL doesn't return a 404, but returns something else that's not valid JSON. (I know we discussed this at some point, and I think it might be already implemented. If not, maybe we add it to the backlog for future updates).

Reply:
This is an important point, and the system does not handle broken JSON as "deleted". I've created an issue for it:
https://github.com/MurmurationsNetwork/MurmurationsServices/issues/266

We'll fix this prior to v2 going into production.